boards/DCM/Test/Src/Test_StateMachine.cpp

```diff
@@ -398,5 +412,172 @@ TEST_F(
         }
     }
 }
+// DCM-8
+TEST_F(DcmStateMachineTest, regen_not_allowed_when_no_airs_closed)
+{
+    SetInitialState(App_GetDriveState());
+
+    // Turn the DIM start switch on to prevent state transitions in
+    // the drive state.
+    App_CanRx_DIM_SWITCHES_SetSignal_START_SWITCH(
+        can_rx_interface, CANMSGS_DIM_SWITCHES_START_SWITCH_ON_CHOICE);
+
+    App_CanRx_FSM_PEDAL_POSITION_SetSignal_MAPPED_PEDAL_PERCENTAGE(
+        can_rx_interface, 60.0f);
+    App_CanRx_DIM_REGEN_PADDLE_SetSignal_MAPPED_PADDLE_POSITION(
+        can_rx_interface, 50.0f);
+
+    float expected_torque_request_value =
+        60.0f / 100.0f * MAX_TORQUE_REQUEST_NM;
+    float value_over_threshold_wheel_speed = std::nextafter(
+        REGEN_WHEEL_SPEED_THRESHOLD_KPH, std::numeric_limits<float>::max());
+
+    // Both wheel speeds over regen threshold
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_LEFT_WHEEL_SPEED(
+        can_rx_interface, value_over_threshold_wheel_speed);
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_RIGHT_WHEEL_SPEED(
+        can_rx_interface, value_over_threshold_wheel_speed);
+    LetTimePass(state_machine, 10);
+
+    // Both AIRs are open
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_NEGATIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_NEGATIVE_OPEN_CHOICE);
+    App_CanRx_BMS_AIR_STATES_SetSignal_AIR_POSITIVE(
+        can_rx_interface, CANMSGS_BMS_AIR_STATES_AIR_POSITIVE_OPEN_CHOICE);
+
+    ASSERT_FLOAT_EQ(
+        expected_torque_request_value,
+        App_CanTx_GetPeriodicSignal_TORQUE_REQUEST(can_tx_interface));
+}
+
+TEST_F(DcmStateMachineTest, regen_not_allowed_when_one_air_closed)
+{
+    SetInitialState(App_GetDriveState());
+
+    // Turn the DIM start switch on to prevent state transitions in
+    // the drive state.
+    App_CanRx_DIM_SWITCHES_SetSignal_START_SWITCH(
+        can_rx_interface, CANMSGS_DIM_SWITCHES_START_SWITCH_ON_CHOICE);
+
+    App_CanRx_FSM_PEDAL_POSITION_SetSignal_MAPPED_PEDAL_PERCENTAGE(
+        can_rx_interface, 60.0f);
+    App_CanRx_DIM_REGEN_PADDLE_SetSignal_MAPPED_PADDLE_POSITION(
+        can_rx_interface, 50.0f);
+
+    float expected_torque_request_value =
+        60.0f / 100.0f * MAX_TORQUE_REQUEST_NM;
+    float value_over_threshold_wheel_speed = std::nextafter(
+        REGEN_WHEEL_SPEED_THRESHOLD_KPH, std::numeric_limits<float>::max());
+
+    // Both wheel speeds over regen threshold
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_LEFT_WHEEL_SPEED(
+        can_rx_interface, value_over_threshold_wheel_speed);
+    App_CanRx_FSM_WHEEL_SPEED_SENSOR_SetSignal_RIGHT_WHEEL_SPEED(
+        can_rx_interface, value_over_threshold_wheel_speed);
+
+    // Case where negative AIR is open
```

Comment:
nit. For these comments it would be nice to write something similar to what you wrote for `    regen_allowed_only_when_going_faster_than_5kph_and_both_airs_closed`

Reply:
ok. Structure of comments is similar now